### PR TITLE
doc: clarify EventSourcedBehavior effects, #25953

### PR DESCRIPTION
**Pull request by patriknw**
* move section up
* example that is using newState parameter

Refs #25953

### CODE REVIEW

File: akka-docs/src/main/paradox/typed/persistence.md [hunk 1]
```diff
@@ -216,6 +212,73 @@ Scala
...
+* `persist` will persist one single event or several events atomically, i.e. all events
+  are stored or none of them are stored if there is an error
+* `none` no events are to be persisted, for example a read-only command
+* `unhandled` the command is unhandled (not supported) in current state
+* `stop` stop this actor
+* `stash` the current command is stashed
+* `unstashAll` process the commands that were stashed with @scala[`Effect.stash`]@java[`Effect().stash`]
+* `reply` send a reply message to the given `ActorRef`
+
+Note that there is only one of these. It is not possible to both persist and say none/unhandled.
```
Comment (johanandren) [inline akka-docs/src/main/paradox/typed/persistence.md]: "Note that only one of those can be chosen per incoming command" or something along those lines?

File: akka-docs/src/main/paradox/typed/persistence.md [hunk 2]
```diff
@@ -216,6 +212,73 @@ Scala
...
+
+Scala
+:  @@snip [PersistentActorCompileOnlyTest.scala](/akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PersistentActorCompileOnlyTest.scala) { #commonChainedEffects }
+
+Java
+:  @@snip [PersistentActorCompileOnlyTest.java](/akka-persistence-typed/src/test/java/akka/persistence/typed/javadsl/PersistentActorCompileOnlyTest.java) { #commonChainedEffects }
+
+### Side effects ordering and guarantees
+
+Any side effects are executed on an at-once basis and will not be executed if the persist fails.
```
Comment (johanandren) [inline akka-docs/src/main/paradox/typed/persistence.md]: Maybe point out that they are not executed on replay but only on original persist, not that it is unclear but can't hurt?

File: akka-docs/src/main/paradox/typed/persistence.md [hunk 3]
```diff
@@ -216,6 +212,73 @@ Scala
...
+The side effects are executed sequentially, it is not possible to execute side effects in parallel.
+
+### Atomic writes
+
+It is possible to store several events atomically by using the `persistAll` effect. That means that all events
```
Comment (johanandren) [inline akka-docs/src/main/paradox/typed/persistence.md]: it's not called `persistAll` it is `persist(list/vararg)`

File: akka-docs/src/main/paradox/typed/persistence.md (hunk 2 above)
Comment (patriknw) [reply akka-docs/src/main/paradox/typed/persistence.md]: added some more here, worth reviewing again

**Review by chbatey — APPROVED** [review]
:+1:

**Review by johanandren — APPROVED** [review]
LGTM

**Pull request merged**